src/format/Token.cc

```diff
@@ -476,6 +538,16 @@ Format::Token::parse(const char *def, Quoting *quoting)
 
     switch (type) {
 
+    case LFT_BYTE:
+        if (!data.string)
+            throw TextException("logformat %byte requires a parameter (e.g., %byte{10})", Here());
+        // TODO: Convert Format::Token::data.string to SBuf.
+        if (const auto v = ParseUnsignedDecimalInteger<uint8_t>("logformat %byte{value}", SBuf(data.string)))
+            data.byteValue = v;
+        else
+            throw TextException("logformat %byte{n} does not support zero n values yet", Here());
+        break;
```

Comment:
This should implement the `%byte{xx}` code requirements and validation fully without needing that controversial template, and adds the missing support for `00` bytes.
```suggestion
    case LFT_BYTE: {
            if (!data.string)
                throw TextException("logformat %byte requires a parameter (e.g., %byte{0A})", Here());
            // TODO: Convert Format::Token::data.string to SBuf.
            Parser::Tokenizer tok(SBuf(data.string));
            int64_t v = 0;
            if (tok.int64(v, 16, false, 2) && tok.atEnd())
                data.byteValue = v;
            else
                throw TextException(ToSBuf("invalid byte value in logformat %byte{", data.string, "}", Here());
            }
        break;
```

Reply:
> This should implement the `%byte{xx}` code requirements and validation fully 

That assertion depends on how those requirements are defined. For example, I support extending %byte interface in the future to support hex numbers, but I would be very much against switching to requirements that match this implementation (now or in the future).

> and adds the missing support for `00` bytes.

While both PR code and the suggested replacement successfully parse the corresponding NUL byte specs, the rest of Squid still does not support NUL bytes in this context. The problem with NULs is in _honoring_ those specs, not _parsing_ them. Otherwise, this PR would have included NUL support! See PR description for why NULs are not supported. Just like with hex, we can add that support later if somebody needs it.

Thank you for clearing this PR despite this disagreement.

P.S. I have polished modified PR description further to fix M-failed-description (and to use a slightly more permanent reference to a known use case).